src/internal/pfssync/cache_client.go

```diff
@@ -66,3 +69,45 @@ func (cc *CacheClient) put(key string, commit *pfs.Commit) {
 	defer cc.mu.Unlock()
 	cc.cache.Add(key, commit)
 }
+
+func (cc *CacheClient) WithCreateFileSetClient(cb func(client.ModifyFile) error) (*pfs.CreateFileSetResponse, error) {
+	pachClient := pfsload.NewPachClient(cc.APIClient)
```

Comment:
Why can't we just use the embedded APIClient?

Reply:
We can but I noticed, the `WithCreateFileSetClient()` function for `pfsload.Client` seems to wrap the `WithCreateFileSetClient()` for `APIClient` and adds a metadata() call to ensure that the metadata is configured properly. If that step is unnecessary then, we can instead use the embedded `APIClient` instead

Reply:
That step is specific to the load testing tooling, the pach client available through the driver is already properly configured.

Reply:
Gotcha, so I will swap back to the `APIClient`